client/rpc-api/src/state/mod.rs

```diff
@@ -136,6 +139,16 @@ pub trait StateApi<Hash> {
 		hash: Option<Hash>
 	) -> FutureResult<Vec<StorageChangeSet<Hash>>>;
 
+	/// Query storage entries (by key) starting at block hash given as the second parameter.
+	#[rpc(name = "state_queryStorageAt")]
+	fn query_storage_at(
+		&self,
+		keys: Vec<StorageKey>,
+		at: Hash,
```

Comment:
I think we should make `Hash` optional, to have consistency with other methods. That would unfortunately mean that you can't use the default implementation and will have to move that logic to impl blocks though.